File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.go
func removeDuplicates(nums []int) int {
    l := len(nums)
    if l < 2 {
        return l
    }

    k := 0
    v := nums[0]
    for i := 1; i < l; i++ {
        if nums[i] != v {
            k++
            v = nums[i]
            if k != i {
                nums[k] = v
            }
        }
    }
    return k+1
}

File: remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.c
int removeDuplicates(int* nums, int numsSize){
    if (numsSize < 2) {
        return numsSize;
    }

    int k = 0, v = nums[0];
    for (int i = 1; i < numsSize; i++) {
        if (nums[i] != v) {
            k++;
            v = nums[i];
            if (k != i) {
                nums[k] = v;
            }
        }
    }

    return k+1;
}


